server/src/main/scala/io/delta/sharing/server/DeltaSharingService.scala

```diff
@@ -396,28 +435,47 @@ class DeltaSharingService(serverConfig: ServerConfig) {
       }
     }
     val responseFormatSet = getResponseFormatSet(capabilitiesMap)
-    val queryResult = deltaSharedTableLoader.loadTable(tableConfig).query(
-      includeFiles = true,
-      request.predicateHints,
-      request.jsonPredicateHints,
-      request.limitHint,
-      request.version,
-      request.timestamp,
-      request.startingVersion,
-      request.endingVersion,
-      request.maxFiles,
-      request.pageToken,
-      request.includeRefreshToken.getOrElse(false),
-      request.refreshToken,
-      responseFormatSet = responseFormatSet)
-    if (queryResult.version < tableConfig.startVersion) {
-      throw new DeltaSharingIllegalArgumentException(
-        s"You can only query table data since version ${tableConfig.startVersion}."
-      )
+
+    if(getAsyncQuery(capabilitiesMap)) {
+      if (!request.idempotencyKey.isDefined) {
+        throw new DeltaSharingIllegalArgumentException(
+          "idempotencyKey is required for async query."
+        )
+      }
+
+      val queryId = s"${share}_${schema}_${table}"
+
+      streamingOutput(
+        Some(0),
+        "parquet",
+        Seq(
+        SingleAction(queryStatus = QueryStatus(queryId, "pending"))
```

Comment:
So this is always returning "pending"?

Reply:
for this demo service, we already return pending if user start async.